.github/workflows/main.yml

```diff
@@ -1,105 +1,59 @@
-# This is a basic workflow to help you get started with Actions
-
 name: CI
 
-# Controls when the action will run. 
-on: [push, pull_request]
-  # Triggers the workflow on push or pull request events but only for the master branch
-  #push:
-  #  branches: [ master ]
-  #pull_request:
-  #  branches: [ master ]
+on:
+  # Runs on every push. If a push contains multiple commits, it will be ran on the latest one.
+  push:
 
-  # Allows you to run this workflow manually from the Actions tab
-  #workflow_dispatch:
+env:
+  COMPILE_JOBS: 2
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  build-release:
-    # The type of runner that the job will run on
+  build:
+    name: Build (${{ matrix.build_type }}-deal.ii:${{ matrix.dealii_version }})
     runs-on: ubuntu-latest
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
-    steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
-
+    strategy:
+      fail-fast: false
+      matrix:
+        build_type: ["Release", "Debug"]
+        dealii_version: ["master", "v9.3.0"]
```

Comment:
I love this. Clearly shows that I have very limited knowledge of Github Actions.